smartfile/__init__.py

```diff
@@ -124,8 +124,40 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        try:
+            return self.post('/path/oper/remove', path=deletefile)
+        except KeyError:
+            raise Exception("Destination file does not exist")
+        
+    def upload(self, usrfile):
+        # needed to split the tuple
+        newtuple = usrfile
+        # splits tuple to allow ability to remove "/" at the end, if present 
+        ourfile, newdata = newtuple
+        if ourfile[-1:] == "/":
+             raise Exception("Can only upload files")
+        return self.post('/path/data/', file=usrfile)  
+        
+    def download(self, downloadfile):
+        return self.get('/path/data/', downloadfile)
+    
+    def move(self, sourcefile, destination):
+        # check destination folder for / at end
+        if destination[-1:] != "/":
+            raise Exception("Destination folder requires a / at end")
+        # check destination folder for / at begining
+        if destination[:-1] != "/":
+            destination = "/" + destination
+        try:
+            t = self.post('/path/oper/move/', src=sourcefile, dst=destination)
+        except KeyError:
+            raise Exception("Destination directory does not exist")
```

Comment:
Mmmm I would expect the server to return a specific exception / error code instead.